### ISSUE TITLE
Yubiswitch doesn't notice Yubikey powering back on

### ISSUE THREAD

**Issue by indolering** [pallotron/yubiswitch#10]
I regularly have issues with Yubiswitch detecting the presence of the Yubikey and also whether it is off.

For example, if I manually remove and reinsert the Yubikey, Yubiswitch does not detect that the Yubikey has been removed and does not power the Yubiswitch back down.  

However, this behavior is occurs without manual intervention.  I will try to take notes (now that I know this repo is actually under development) but I suspect that it may be linked to #5.

Perhaps there are some system hooks that get called when activating USB devices that you could listen in on?

**Comment by aerickson** [comment]
I have issues with my YubiKey becoming enabled again (despite the checkbox not being enabled). I notice it when I brush up against it and I paste a OTP into chat. :)

I am running FDE...

**Comment by sixninetynine** [comment]
+1, FDE here as well.

**Comment by pallotron** [comment]
can you try version 0.9 (which I just pushed) and let me know if you still have similar problems?

**Comment by pallotron** [comment]
have you got a chance to try 0.9?

**Comment by aerickson** [comment]
I just tried 0.9. I can't get the Yubikey to emit an OTP when enabled. 

I didn't restart... is that required? I killed the daemon (kill -9 required... it gets hung up) - /Library/PrivilegedHelperTools/com.pallotron.yubiswitch.helper
-  (it didn't exit after quitting Yubiswitch, is it supposed to?), but that didn't help.

I've killed the daemon, quit the app, and removed/inserted the Yubikey and I still can't get it to emit an OTP.

**Comment by pallotron** [comment]
> I just tried 0.9. I can't get the Yubikey to emit an OTP when enabled.

weird, it works for me. what yubikey do you have? have you verified that productID and vendorID strings are set correctly in the preferences? you can get the value from the system report (looking at USB section)

> I didn't restart... is that required? I killed the daemon (kill -9 required... it gets hung up) - /Library/PrivilegedHelperTools/com.pallotron.yubiswitch.helper

shouldn't hang, mine dies when you sent normal SIGTERM signal

> (it didn't exit after quitting Yubiswitch, is it supposed to?), but that didn't help.

yeah, helpers usually don't exit with the main application

> I've killed the daemon, quit the app, and removed/inserted the Yubikey and I still can't get it to emit an OTP.

:(

what do you see in /var/log/system.log? grepping for "yubi"

**Comment by aerickson** [comment]
I have a Nano, the System Information was strange... it just listed the key as 'Composite Device'. After pulling it out and putting it back in a few times and still not working I cleaned the contacts... there was some dirt. It's now showing in SI as 'Yubico Yubikey II' and it's behaving properly.

I've tested by putting the machine to sleep and then touching it - no key was emitted. After enabling and touching a key is emitted.

I think the fix works. 

Thanks!
